go/cmd/vtbench/vtbench.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2018 The Vitess Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	"vitess.io/vitess/go/exit"
+	"vitess.io/vitess/go/vt/dbconfigs"
+	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/logutil"
+	"vitess.io/vitess/go/vtbench"
+
+	_ "vitess.io/vitess/go/vt/vtgate/grpcvtgateconn"
+	_ "vitess.io/vitess/go/vt/vttablet/grpctabletconn"
+)
+
+var (
+	// connection flags
+	host     = flag.String("host", "", "vtgate host(s) in the form 'host1,host2,...'")
+	port     = flag.Int("port", 0, "vtgate port")
+	protocol = flag.String("protocol", "mysql", "client protocol, either mysql (default) or grpc")
+	user     = flag.String("user", "", "username to connect using mysql (password comes from the db-credentials-file)")
+	db       = flag.String("db", "", "db name to use when connecting / running the queries (e.g. @replica, keyspace, keyspace/shard etc)")
+
+	// test flags
+	deadline = flag.Duration("deadline", 5*time.Minute, "maximum duration for the test run (default 5 minutes)")
+	sql      = flag.String("sql", "", "sql statement to execute")
+	threads  = flag.Int("threads", 2, "number of parallel threads to run")
+	count    = flag.Int("count", 1000, "number of queries per thread")
+)
+
+func main() {
+	logger := logutil.NewConsoleLogger()
+	flag.CommandLine.SetOutput(logutil.NewLoggerWriter(logger))
+
+	defer exit.Recover()
+
+	flag.Lookup("logtostderr").Value.Set("true")
+	flag.Parse()
+
+	clientProto := vtbench.MySQL
+	switch *protocol {
+	case "", "mysql":
+		clientProto = vtbench.MySQL
+	case "grpc-vtgate":
+		clientProto = vtbench.GRPCVtgate
+	case "grpc-vttablet":
+		clientProto = vtbench.GRPCVttablet
```

Comment:
Is this to run the test directly against the tablet? We should include it in the protocol list doc string.

Reply:
Good idea. Done